### PR TITLE
Jetpack Cloud: Add thank you note to JP Scan purchases

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

- We need to add a "Thank you" note after purchasing Jetpack Scan.

#### Important note

It is still unclear what should the CTA do. Right now, it brings the user to the checklist tour. This seems reasonable since Scan requires the server credentials to work and we could add that requirement to the checklist.

#### Testing instructions

- Navigate to http://calypso.localhost:3000/plans/
- Purchase Jetpack Scan
- You should see the "Thank you" note and the end of the URL should look like `?thank-you=&product=jetpack_scan`

See 1151678672052943-as-1168017254149561.

#### Demo
Kapture 2020-03-30 at 15 01 54

### CODE REVIEW

**Comment by enejb** [comment]
@keoshi Some questions. 
Where should the button take you?
Any other design feedback?

**Comment by keoshi** [comment]
@enejb Yesterday I suggested we could have two buttons:

`[Add server credentials now] [See checklist]`

- The first one, which is the CTA, would take you to settings and [highlight the server credentials](https://github.com/Automattic/wp-calypso/pull/40018) section.
- The second one would simply close the modal and let the user browse through the checklist.

p1585587050117500-slack-jetpack-voyager

Spinning the PR to test it fully.

**Comment by singerb** [comment]
We do have some fixers currently, with plans to have more; what we call them is more a marketing thing. I'm happy with "one-click" but we've used some form of "automated" as well before.

**Comment by keoshi** [comment]
Thanks for confirming that, @singerb ! Appreciate it.

**Comment by rcanepa** [comment]
Hey @keoshi ! 

> The modal width is slight inconsistent with others. I assume this has to do with the amount of content inside that automatically resizes the modal up to a max-width (that's what I got from the quick code poking I did). Can we do anything to make these more consistent? See comparison shots below.

You are right. Depending on the content, the icon/image of the left takes different amounts of space making the modal height greater in some cases, and that makes the whole modal narrower. Notice how the images differ in size in each case.

ThankYouJPSearch
ThankYouJPBackups
ThankYouJPScan

If we set a max-width in pixels to the image, we could have more consistency between the three modals. What do you think?

> We could use the same illustration that's in Backup right now — I actually think it fits Scan better than it fits the Backup product. But again, I don't mind the trophy because it's more celebratory and that what we'll be using in Search too.

Sure. I prefer the security image/icon as well. I'm pushing a new change with it (I rebased the PR by the way).

**Comment by rcanepa** [comment]
I pushed a new version in which I set a `max-width` for the image and force the `width` of the modal to `720px` when `breakpoint( '>800px' )`.

Kapture 2020-03-31 at 16 11 40

**Comment by keoshi** [comment]
Oh, actually, one more thing: did we ever change the [buttons and their actions](https://github.com/Automattic/wp-calypso/pull/40580#issuecomment-606539803)? I'm not seeing that on the code or the latest screenshots.

**Comment by rcanepa** [comment]
@keoshi . I pushed a new change that includes the `[Add server credentials now] [See checklist]` buttons. Let me know what you think!
image

**Comment by keoshi** [comment]
@rcanepa Can we just remove the `primary` property from the second button so there's only one clear primary action? Other than that it looks awesome to me! Thank you once again.

**Comment by rcanepa** [comment]
Done it! 
image

**Comment by keoshi** [comment]
Perfect, let’s ship it.

File: client/my-sites/checkout/checkout/index.jsx [hunk 1]
```diff
@@ -38,7 +38,11 @@ import {
 	hasTransferProduct,
 	jetpackProductItem,
 } from 'lib/cart-values/cart-items';
-import { JETPACK_BACKUP_PRODUCTS, JETPACK_SEARCH_PRODUCTS } from 'lib/products-values/constants';
+import {
+	JETPACK_BACKUP_PRODUCTS,
```
Comment (enejb) [inline client/my-sites/checkout/checkout/index.jsx:42]: Instead of always having to import a new product constant we can use `isJetpackProductSlug`

**Comment by enejb** [comment]
This worked as expected for me! 

Nicely done. 

One thing that we should address in a different PR is that we currently don't show the Remote credentials form for sites with jetpack scan. 

Lets ship this!

**Comment by rcanepa** [comment]
> One thing that we should address in a different PR is that we currently don't show the Remote credentials form for sites with jetpack scan.

I would love to work on that.

> Lets address the one comment that I made and ship this.

Done! I'll wait for your approval to ship it @enejb .

**Pull request merged**